Code/Framework/AzCore/AzCore/Math/MatrixMxN.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Math/MatrixMxN.h>
+#include <AzCore/Math/MathUtils.h>
+#include <AzCore/Math/MathScriptHelpers.h>
+
+namespace AZ
+{
+    void MatrixMxN::Reflect([[maybe_unused]] ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<SerializeContext*>(context))
+        {
+            serializeContext->Class<MatrixMxN>()
+                ->Version(1)
+                ->Field("RowCount", &MatrixMxN::m_rowCount)
+                ->Field("ColCount", &MatrixMxN::m_colCount)
+                ->Field("RowGroups", &MatrixMxN::m_numRowGroups)
+                ->Field("ColGroups", &MatrixMxN::m_numColGroups)
+                ->Field("Values", &MatrixMxN::m_values)
+                ;
+
+            if (EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<MatrixMxN>("N-Dimensional Vector", "")
+                    ->ClassElement(Edit::ClassElements::EditorData, "")
+                    ->DataElement(
+                        Edit::UIHandlers::Default, &MatrixMxN::m_rowCount, "Total Rows", "The total number of rows in the matrix")
+                    ->DataElement(
+                        Edit::UIHandlers::Default, &MatrixMxN::m_colCount, "Total Columns", "The total number of columns in the matrix")
```

Comment:
These should probably have ChangeNotify handlers that resizes m_values, m_numRowGroups, and m_numColGroups whenever they change? Otherwise the values could all get out of sync.